src/components/Cards/CollectionCard.vue

```diff
@@ -0,0 +1,248 @@
+                  >
+                    {{ $t('quest-card:play') }}
+                  </b-button>
+                  <SmartLink v-if="started" :link="toGame">
+                    <b-progress :value="to_next()" max="1"></b-progress>
+                  </SmartLink>
+                </div>
+              </div>
+            </div>
+          </b-col>
+        </b-row>
+      </template>
+    </AspectRatioCard>
+
+  </SmartLink>
+</template>
+<script lang="ts">
+  import { Component, Mixins, Prop, Ref, Vue } from 'vue-property-decorator';
+  import AspectRatioCard from '@/components/Cards/AspectRatioCard.vue';
+  import { PUZZLE_ROUTE_PREFIX } from '@/utils/constants';
+  import SmartLink from '@/components/Common/SmartLink.vue';
+  import Utils from '@/utils/utils';
+  import {
+    ClearedPuzzle,
+    CollectionResponse,
+    PuzzleItem,
+  } from '@/types/common-types';
+  import FetchMixin from '@/mixins/FetchMixin';
+
+  @Component({
+    components: {
+      AspectRatioCard,
+      SmartLink,
+    },
+  })
+  export default class CollectionCard extends Mixins(FetchMixin) {
+    @Prop({ required: true }) readonly image!: string;
+
+    @Prop({ required: true }) readonly title!: string;
+
+    @Prop({ required: true }) readonly desc!: string;
+
+    @Prop({ required: false }) readonly level!: string;
+
+    @Prop({ required: true }) readonly nid!: string;
+
+    @Prop({ required: true }) readonly founder_picture!: string;
+
+    @Prop({ required: true }) readonly founder_name!: string;
+
+    @Prop({ required: true }) readonly puzzles!: string;
+
+    @Prop({ required: true }) readonly cleared!: PuzzleItem[];
+
+    @Prop({ default: 1 }) readonly aspectRatio!: number;
+
+    @Prop() readonly collectionLink?: string;
+
+    @Prop() readonly puzzleLink?: string;
+
+    @Prop({ required: false }) readonly current_level!: string;
+
+    @Prop() readonly current_puzzle?: string;
+
+    @Ref('root') readonly root!: HTMLDivElement;
+
+    puzzleList: String[] = [];
+
+    get nav() {
+      return Utils.isLinkInternal(this.toGame) ? 'to' : 'href';
+    }
+
+    get toGame() {
+      return (
+        this.puzzleLink ||
+        (this.current_puzzle &&
+          `${PUZZLE_ROUTE_PREFIX}${this.current_puzzle}/`) ||
+        this.toCollection
+      );
+    }
+
+    get toCollection() {
+      return this.collectionLink || (this.title && `/collections/${this.nid}`);
+    }
+
+    get locked() {
+      return Number(this.level) - 1 > Number(this.current_level);
+    }
+
+    get started() {
+      return this.to_next() > 0 && !this.locked;
+    }
+
+    get completed() {
+      return this.to_next() >= 1 && !this.locked;
+    }
+
+    to_next() {
+      const cleared = this.cleared.filter((x) =>
+        this.puzzleList.map((y) => y).includes(x.nid),
+      );
+      return cleared.length / this.puzzleList.length;
+    }
+
+    async fetch() {
+      this.puzzleList = this.puzzles.split(',');
+    }
+
+    getImage(nid: string) {
```

Comment:
Can this be a getter since it's only called with the first entry of the puzzles array?